client-spark/common/src/main/java/org/apache/spark/shuffle/reader/RssShuffleDataIterator.java

```diff
@@ -166,7 +166,7 @@ private int uncompress(CompressedShuffleBlock rawBlock, ByteBuffer rawData) {
           || !isSameMemoryType(uncompressedData, rawData)) {
 
         if (LOG.isDebugEnabled()) {
-          if (!isSameMemoryType(uncompressedData, rawData)) {
+          if (uncompressedData != null && !isSameMemoryType(uncompressedData, rawData)) {
```

Comment:
`rawData != null`

Reply:
`rawData` will not be null. Because the code is something like:
```
public boolean hasNext() {
      ...
      if (rawData != null) {
        uncompress(rawBlock, rawData);
      }
}
```

Reply:
OK.